gap/base/recognition.gi

```diff
@@ -678,18 +678,19 @@ InstallGlobalFunction( CalcNiceGensGeneric,
 InstallGlobalFunction( CalcNiceGensHomNode,
   # function for the situation on a homomorphism node (non-Leaf):
   function(ri,origgens)
-    local origkergens,rifac,riker,pregensfactor;
+    local gensOfKernel,rifac,riker,preimagesOfNiceGensOfFactor,
+    niceGensOfKernel;
     # Is there a non-trivial kernel?
     rifac := RIFac(ri);
+    preimagesOfNiceGensOfFactor := CalcNiceGens(rifac,origgens);
     if HasRIKer(ri) and RIKer(ri) <> fail then
-        pregensfactor := CalcNiceGens(rifac,origgens);
         riker := RIKer(ri);
-        origkergens := ResultOfStraightLineProgram( gensNslp(ri), origgens );
-        return Concatenation( pregensfactor,
-                              CalcNiceGens(riker,origkergens) );
+        gensOfKernel := ResultOfStraightLineProgram( gensNslp(ri), origgens );
```

Comment:
Looking at the def for `gensNslp`,  I saw this:
```
        SetgensNslp(ri,SLPOfElms(gensN(ri)));
```
Which made me wonder why one can't just use something like `List( gensN(ri), StripMemory )` here..

The answer being (I think) that the output here really needs to depend on `origgens`, and these might not be original gens we worked, they could be arbitrary stand-ins which satisfy the same relations. Perhaps this could be made clearer somehow by some comments?